lib/Raven/Client.php

```diff
@@ -112,13 +119,27 @@ class Client
     protected $config;
 
     /**
+     * @var FlexibleHttpClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory The PSR-7 request factory
+     */
+    private $requestFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration      $config         The client configuration
+     * @param FlexibleHttpClient $httpClient     The HTTP client
+     * @param RequestFactory     $requestFactory The PSR-7 request factory
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, FlexibleHttpClient $httpClient, RequestFactory $requestFactory)
```

Comment:
I know, but I found shocking that there is not an interface that both implements `HttpClient` and `HttpAsyncClient`. The only thing that I found was that class, and our requirements are to support both syncronous and asyncronous requests (and clients, obviously). I don't like this either 😞